AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Sync/SubscriptionSync/AWSIncomingEventReconciliationQueue.swift

```diff
@@ -28,21 +28,26 @@ final class AWSIncomingEventReconciliationQueue: IncomingEventReconciliationQueu
         return eventReconciliationQueueTopic.eraseToAnyPublisher()
     }
 
+    private let connectionStatusSerialQueue: DispatchQueue
     private var reconciliationQueues: [String: ModelReconciliationQueue]
     private var reconciliationQueueConnectionStatus: [String: Bool]
+    private var modelReconciliationQueueFactory: ModelReconciliationQueueFactory
 
     init(modelTypes: [Model.Type],
          api: APICategoryGraphQLBehavior,
-         storageAdapter: StorageEngineAdapter) {
+         storageAdapter: StorageEngineAdapter,
+         modelReconciliationQueueFactory: ModelReconciliationQueueFactory? = nil) {
         self.modelReconciliationQueueSinks = [:]
         self.eventReconciliationQueueTopic = PassthroughSubject<IncomingEventReconciliationQueueEvent, DataStoreError>()
         self.reconciliationQueues = [:]
         self.reconciliationQueueConnectionStatus = [:]
+        self.modelReconciliationQueueFactory = modelReconciliationQueueFactory ??
+            AWSModelReconciliationQueue.init(modelType:storageAdapter:api:incomingSubscriptionEvents:)
+        //TODO: Add target for SyncEngine system to help prevent thread explosino and increase performance
```

Comment:
Unless this is going to be addressed in an imminent PR, consider adding an issue to our backlog, assigned to the appropriate milestone, and linking it here so we don't lose track of this.

Reply:
Issue opened #399